PROTOCOL.md

```diff
@@ -564,22 +565,54 @@ The following is an example `rowIdHighWaterMark` action:
 }
 ```
 
+### Domain Metadata
+The domain metadata action contains a configuration (string-string map) for a named metadata domain. Two overlapping transactions conflict if they both contain a domain metadata action for the same metadata domain.
+
+There are two types of metadata domains:
+1. **User-controlled metadata domains** have names that start with anything other than the `delta.` prefix. Any Delta client implementation or user application can modify these metadata domains, and can allow users to modify them arbitrarily. Delta clients and user applications are encouraged to use a naming convention designed to avoid conflicts with other clients' or users' metadata domains (e.g. `com.databricks.*` or `org.apache.*`).
+2. **System-controlled metadata domains** have names that start with the `delta.` prefix. Only Delta client implementations are allowed to modify the metadata for system-controlled domains. A Delta client implementation should only update metadata for system-controlled domains that it knows about and understands. System-controlled metadata domains are used by various table features and each table feature may impose additional semantics on the metadata domains it uses. `delta.` prefix is reserved for metadata domains mentioned in the Delta spec (e.g. as part of some table feature).
```

Comment:
> A Delta client implementation should only update metadata for system-controlled domains that it knows about and understands

How should clients handle a system metadata domain they don't understand? Are system metadata domains _always_ associated with some table feature that would block writing and/or reading the table? Or should the client basically just treat it like a user metadata domain at that point?

Reply:
I could imagine system-controlled domains for which the default domain handling suffices (writers don't erase or modify its content), but which are documented in the Delta spec and thus can be manipulated by a client who actually understands them? This would cover any kind of auxiliary/optional information that isn't important enough to force the compat requirements that come with a table feature, but where it would still be nice if other readers didn't actively destroy the domain's info? Maybe something like query optimizer stats that don't need to be perfectly up to date, so it's better to let an older/unaware client to propagate the existing (now stale) entry,  rather than block that client or delete the stats?